### PR TITLE
[tests] add test steps to verify case-insensitive query works

### PR DESCRIPTION
**Pull request by simonlingoogle**

### CODE REVIEW

**Review by abtink — APPROVED** [review]
LGTM. Thanks. 👍 

In https://github.com/openthread/openthread/pull/7189, `test_dnssd.py` was updated to  check the case-insensetive behavior (part of the last commit). I agree it is good to also cover it with names containing space and/or special chars.

File: tests/scripts/thread-cert/border_router/test_dnssd_instance_name_with_space.py [hunk 1]
```diff
@@ -115,6 +115,7 @@ def test(self):
 
         self._verify_service_browse_result(client.dns_browse(SERVICE_FULL_NAME, server=br1.get_rloc()))
         self._verify_service_browse_result(client.dns_browse(SERVICE_FULL_NAME, server=br2.get_rloc()))
+        self._verify_service_browse_result(client.dns_browse(SERVICE_FULL_NAME.lower(), server=br2.get_rloc()))
```
Comment (jwhui) [inline tests/scripts/thread-cert/border_router/test_dnssd_instance_name_with_space.py:118]: What is the purpose of this? Is `SERVICE_FULL_NAME` already all lower case?

Comment (simonlingoogle) [reply tests/scripts/thread-cert/border_router/test_dnssd_instance_name_with_space.py:118]:
```suggestion
        self._verify_service_browse_result(client.dns_browse(SERVICE_FULL_NAME.upper(), server=br2.get_rloc()))
```

**Comment by jwhui** [comment]
@simonlingoogle , this check seems to be failing intermittently on this PR:
```
======================================================================
FAIL: test (__main__.TestDnssdInstanceNameWithSpace)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "./tests/scripts/thread-cert/border_router/test_dnssd_instance_name_with_space.py", line 118, in test
    self._verify_service_browse_result(client.dns_browse(SERVICE_FULL_NAME.upper(), server=br2.get_rloc()))
  File "./tests/scripts/thread-cert/border_router/test_dnssd_instance_name_with_space.py", line 217, in _verify_service_browse_result
    self.assertEqual(service['host'], HOST_FULL_NAME)
AssertionError: 'host1.DEFAULT.SERVICE.ARPA.' != 'host1.default.service.arpa.'
- host1.DEFAULT.SERVICE.ARPA.
+ host1.default.service.arpa.
----------------------------------------------------------------------
```
https://github.com/openthread/openthread/runs/4320004954?check_suite_focus=true

Thoughts?

**Comment by simonlingoogle** [comment]
> @simonlingoogle , this check seems to be failing intermittently on this PR:
> 
> ```
> ======================================================================
> FAIL: test (__main__.TestDnssdInstanceNameWithSpace)
> ----------------------------------------------------------------------
> Traceback (most recent call last):
>   File "./tests/scripts/thread-cert/border_router/test_dnssd_instance_name_with_space.py", line 118, in test
>     self._verify_service_browse_result(client.dns_browse(SERVICE_FULL_NAME.upper(), server=br2.get_rloc()))
>   File "./tests/scripts/thread-cert/border_router/test_dnssd_instance_name_with_space.py", line 217, in _verify_service_browse_result
>     self.assertEqual(service['host'], HOST_FULL_NAME)
> AssertionError: 'host1.DEFAULT.SERVICE.ARPA.' != 'host1.default.service.arpa.'
> - host1.DEFAULT.SERVICE.ARPA.
> + host1.default.service.arpa.
> ----------------------------------------------------------------------
> ```
> 
> https://github.com/openthread/openthread/runs/4320004954?check_suite_focus=true
> 
> Thoughts?

The tests need fixing. Will fix.

**Comment by simonlingoogle** [comment]
The tests pass now.

**Review by jwhui — APPROVED** [review]

**Pull request merged**